ovs: remove ovs-port- prefix for OVS bridge LAG ports

rhrazdil opened the pull request:
The ovnkube-node pods perform health checks, and they seem to expect the port to be called the same
as the underlying interface:
https://github.com/ovn-org/ovn-kubernetes/blob/acfc09e8a7ffae73918bece102d0c520e4eb48fe/go-controller/pkg/node/healthcheck.go#L298

Renaming of the Ports causes these pods to crash.

Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

ffmancera commented:
/ok-to-test

cathay4t commented:
@rhrazdil Is `ovnkube` checking the NetworkManager `connection.id` or `connection.interface-name`.

Can we use `ovs-port-` prefix for `connection.id` and user desired interface name as `connection.interface-name`?

rhrazdil commented:
Hello @cathay4t 
By the error message in the ovnkube pod it's checking with ovs-vsctl:
```
I1101 14:47:29.362491 2671168 ovs.go:213] exec(16): stderr: "ovs-vsctl: no row \"ovs-port-eno1\" in table Interface\n"
I1101 14:47:29.362495 2671168 ovs.go:215] exec(16): err: exit status 1
F1101 14:47:29.362522 2671168 ovnkube.go:130] Bridge for interface failed for br-ex: Failed to find intfName for br-ex: failed to get ofport of ovs-port-eno1, stderr: "ovs-vsctl: no row \"ovs-port-eno1\" in table Interface\n", error: exit status 1
```

This part specifically: `"ovs-vsctl: no row \"ovs-port-eno1\" in table Interface\n` brings me to the conclusion that they expect the Port name to be the same as the Interface.


> Can we use ovs-port- prefix for connection.id and user desired interface name as connection.interface-name?
Pardon me, but I don't follow, could you elaborate?

EDIT:
> Can we use ovs-port- prefix for connection.id and user desired interface name as connection.interface-name?

If this means that the Port names in ovsdb would be preserved, then I think yes

cathay4t commented:
In that case, please change `nm/connection.py`: `create_new_nm_simple_conn()` to use `-port` postfix for OVS port as connection name, and use user defined name as interface name.

rhrazdil commented:
> In that case, please change nm/connection.py: create_new_nm_simple_conn() to use -port postfix for OVS port as connection name, and use user defined name as interface name.

Thanks! I'm hitting an issue with, when I add the `-port` postfix in `create_new_nm_simple_conn()`,
in this test: `test_remove_ovs_internal_iface_got_port_profile_removed`

The connection has the postfix
```
DEVICE    TYPE           STATE         CONNECTION    
brtest0   ovs-bridge     connected     brtest0-br    
ovstest0  ovs-interface  connected     ovstest0-if   
ovstest0  ovs-port       connected     ovstest0-port 
```
And the bridge has the interfaces unchanged
```
d26c2cae-4823-42b7-8ca9-deb75b276631
    Bridge "brtest0"
        Port "ovstest0"
            Interface "ovstest0"
                type: internal
    ovs_version: "2.11.0"
```

But the test takes the interface names from the bridge connections: https://github.com/nmstate/nmstate/blob/base/tests/integration/nm/ovs_test.py#L122
which contains `['eth1', 'ovstest0']`, so the test fails on this [assert](https://github.com/nmstate/nmstate/blob/base/tests/integration/nm/ovs_test.py#L124)
because of th missing postfix.

> and use user defined name as interface name.

Do you mean setting the name of the interface (not the ovs-port) to be called with the `-port`
postfix as well? Where would be the right place to make this change please?

Commit:
ovs: remove ovs-port prefix

Prepending the prefix causes ovnkube-node pods to crash.
This change adds -port postfix to ovs-interface connection instead.

Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

rhrazdil commented:
@cathay4t Hey, updated the PR according to your suggestion.

Failure in pull-nmstate-integ_tier1-k8s seems to be unrelated as it fails on `base` branch as well

ffmancera reviewed (approved):
LGTM! Could you take a look @cathay4t?

cathay4t commented:
Looks good to me.

cathay4t committed:
Merge branch 'base' into remove-ovs-port-prefix
